x/stakeibc/keeper/hooks.go

```diff
@@ -17,42 +17,64 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochN
 		k.Logger(ctx).Info(fmt.Sprintf("Stride Epoch %d", epochNumber))
 		depositInterval := int64(k.GetParam(ctx, types.KeyDepositInterval))
 		if epochNumber%depositInterval == 0 {
+			// track the current deposit epoch number in stakeibc
+			if epochNumber < 0 {
+				panic("epochs must be positive")
+			}
+			epochTracker := types.EpochTracker{
+				EpochIdentifier: epochIdentifier,
+				EpochNumber: uint64(epochNumber),
+			}
+			// deposit records *must* exist for this epoch
+			k.SetEpochTracker(ctx, epochTracker)
 			// TODO TEST-72 move this function to the keeper
 			k.Logger(ctx).Info("Triggering deposits")
+
+			// Create one new deposit record / host zone for the next epoch
+			createDepositRecords := func(index int64, zoneInfo types.HostZone) (stop bool) {
+				// create a deposit record / host zone
+				depositRecord := types.NewDepositRecord(0, zoneInfo.HostDenom, zoneInfo.ChainId, types.DepositRecord_TRANSFER, uint64(epochNumber))
+				k.AppendDepositRecord(ctx, *depositRecord)
+				return false
+			}
+			// Iterate the zones and apply icaReinvest
+			k.IterateHostZones(ctx, createDepositRecords)
```

Comment:
to make sure I'm tracking: the intent is to create a deposit record per epoch, even if no deposits occur in that epoch?

Reply:
Yes that's right. We might need some edge case logic to handle empty records. We could create a record the first time a user liquid stakes in an epoch, but that would require us to change the way records are stored and use a deterministic key (right now the key for a new record is just an id, which is the deposit record count+1)